src/NuGet.Clients/NuGet.PackageManagement.UI/Models/Package/EmbeddedResourcesCapability.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+#nullable enable
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.PackageManagement.UI.Models.Package
+{
+    internal class EmbeddedResourcesCapability : IEmbeddedResources
+    {
+        private INuGetPackageFileService _nugetPackageFileService;
+        private PackageModel _package;
```

Comment:
Consider passing the Identity instead of the package.

A package will be composed of a `EmbeddedResourcesCapability` so depending on the PackageModel itself makes a circular dependency.
Also, we only need the Identity from the package so being more specific about the dependencies needed helps keep the separation clear.